bundles/org.openhab.binding.bluetooth.govee/src/test/java/org/openhab/binding/bluetooth/govee/internal/readme/ThingTypeTableGenerator.java

```diff
@@ -0,0 +1,118 @@
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathFactory;
+
+import org.apache.commons.lang.StringUtils;
+import org.openhab.binding.bluetooth.govee.internal.GoveeModel;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ * @author Connor Petty - Initial contribution
+ *
+ */
+public class ThingTypeTableGenerator {
+
+    public static void main(String[] args) throws Exception {
+
+        FileInputStream fileIS = new FileInputStream("src/main/resources/OH-INF/thing/thing-types.xml");
+        DocumentBuilderFactory builderFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = builderFactory.newDocumentBuilder();
+        Document xmlDocument = builder.parse(fileIS);
+        XPath xPath = XPathFactory.newInstance().newXPath();
+        String expression = "/*[local-name()='thing-descriptions']/thing-type";
+        XPathExpression labelExpression = xPath.compile("label/text()");
+        XPathExpression descriptionExpression = xPath.compile("description/text()");
+
+        NodeList nodeList = (NodeList) xPath.compile(expression).evaluate(xmlDocument, XPathConstants.NODESET);
+
+        List<ThingTypeData> thingTypeDataList = new ArrayList<>();
+
+        for (int i = 0; i < nodeList.getLength(); i++) {
+            Node node = nodeList.item(i);
+            ThingTypeData data = new ThingTypeData();
+
+            data.id = node.getAttributes().getNamedItem("id").getTextContent();
+            data.label = (String) labelExpression.evaluate(node, XPathConstants.STRING);
+            data.description = (String) descriptionExpression.evaluate(node, XPathConstants.STRING);
+
+            thingTypeDataList.add(data);
+        }
+
+        String[] headerRow = new String[] { "Thing Type ID", "Description", "Supported Models" };
+
+        List<String[]> rows = new ArrayList<>();
+        rows.add(headerRow);
+        rows.addAll(thingTypeDataList.stream().map(ThingTypeTableGenerator::toRow).collect(Collectors.toList()));
+
+        int[] maxColumns = { maxColumnSize(rows, 0), maxColumnSize(rows, 1), maxColumnSize(rows, 2) };
+
+        StringWriter writer = new StringWriter();
+
+        // write actual rows
+        rows.forEach(row -> {
+            writer.append(writeRow(maxColumns, row, ' ')).append('\n');
+            if (row == headerRow) {
+                writer.append(writeRow(maxColumns, new String[] { "", "", "" }, '-')).append('\n');
+            }
+        });
+
+        System.out.println(writer.toString());
+    }
+
+    private static String writeRow(int[] maxColumns, String[] row, char paddingChar) {
+        String prefix = "|" + paddingChar;
+        String infix = paddingChar + "|" + paddingChar;
+        String suffix = paddingChar + "|";
+
+        return Stream.of(0, 1, 2).map(i -> StringUtils.rightPad(row[i], maxColumns[i], paddingChar))
```

Comment:
Maybe not use apache `StringUtils` as we want to get ride of it.

Reply:
done